py/server/deephaven/updateby_clause.py

```diff
@@ -0,0 +1,269 @@
+                   ema_control: EmaControl = None) -> UpdateByClause:
+    """Creates an EMA (exponential moving average) UpdateByClause for the supplied column name pairs, using ticks as
+    the decay unit.
+
+    The formula used is
+        a = e^(-1 / timeScaleTicks)
+        ema_next = a * ema_last + (1 - a) * value
+
+    Args:
+        time_scale_ticks (int): the decay rate (tau) in ticks
+        pairs (Union[str, List[str]]): the input/output column name pairs
+        ema_control (EmaControl): defines how special cases should behave,  when None, the default EmaControl settings
+            as specified in :meth:`~EmaControl.__init__` will be used
+
+    Returns:
+        an UpdateByClause
+
+    Raises:
+        DHError
+    """
+    try:
+        pairs = to_sequence(pairs)
+        if ema_control is None:
+            return UpdateByClause(j_updateby_clause=_JUpdateByClause.Ema(time_scale_ticks, *pairs))
+        else:
+            return UpdateByClause(
+                j_updateby_clause=_JUpdateByClause.Ema(ema_control.j_ema_control, time_scale_ticks, *pairs))
+    except Exception as e:
+        raise DHError(e, "failed to create a tick-decay EMA UpdateByClause.") from e
+
+
+def ema_time_decay(ts_col: str, time_scale_nanos: int, pairs: Union[str, List[str]],
+                   ema_control: EmaControl = None) -> UpdateByClause:
+    """Creates an EMA(exponential moving average) UpdateByClause for the supplied column name pairs, using time as
+    the decay unit.
+
+    The formula used is
+        a = e^(-dt / timeScaleNanos)
+        ema_next = a * ema_last + (1 - a) * value
+
+     Args:
+        ts_col (str): the column in the source table to use for timestamps
+        time_scale_nanos (int): the decay rate (tau) in nanoseconds
+        pairs (Union[str, List[str]]): the input/output column name pairs
+        ema_control (EmaControl): defines how special cases should behave,  when None, the default EmaControl settings
+            as specified in :meth:`~EmaControl.__init__` will be used
+
+    Returns:
+        an UpdateByClause
+
+    Raises:
+        DHError
+     """
+    try:
+        pairs = to_sequence(pairs)
+        if ema_control is None:
+            return UpdateByClause(j_updateby_clause=_JUpdateByClause.Ema(ts_col, time_scale_nanos, *pairs))
+        else:
+            return UpdateByClause(
+                j_updateby_clause=_JUpdateByClause.Ema(ema_control.j_ema_control, ts_col, time_scale_nanos, *pairs))
+    except Exception as e:
+        raise DHError(e, "failed to create a time-decay EMA UpdateByClause.") from e
+
+
+def cum_sum(pairs: Union[str, List[str]]) -> UpdateByClause:
+    """Creates a CumSum (cumulative sum) update-by clause for the supplied column name pairs.
+
+    Args:
+        pairs (Union[str, List[str]]): the input/output column name pairs; when empty, update_by perform the
+            cumulative sum operation on all the applicable columns.
```

Comment:
```suggestion
        pairs (Union[str, List[str]]): the input/output column name pairs; when empty, update_by performs the
            cumulative sum operation on all the applicable columns.
```
Problem is repeated in many places.